src/Extensions/Diagnostics.HealthChecks/Internal/RestHealthCheckPublisher.cs

```
@@ -0,0 +1,112 @@
+				sfh.HealthState.Error => sfc.HealthState.Error,
+				sfh.HealthState.Unknown => sfc.HealthState.Unknown,
+				_ => throw new ArgumentException($"'{healthInfo.HealthState}' is not a valid health status."),
+			};
+
+			sfc.HealthInformation sfcHealthInfo = new(
+				   sourceId: healthInfo.SourceId,
+				   property: healthInfo.Property,
+				   healthState: healthState,
+				   timeToLiveInMilliSeconds: healthInfo.TimeToLive,
+				   description: healthInfo.Description,
+				   sequenceNumber: healthInfo.SequenceNumber.ToString(),
+				   removeWhenExpired: healthInfo.RemoveWhenExpired,
+				   healthReportId: healthInfo.HealthReportId
+		   );
+
+			return sfcHealthInfo;
+		}
+
+		private async Task PublishHealthInfo(sfh.HealthInformation healthInfo, CancellationToken cancellationToken)
+		{
+			IExecutionContext ctx = new BaseExecutionContext();
+			IPAddress addr = ctx.ClusterIpAddress;
+
+			sfc.HealthInformation sfcHealthInfo = fromSfHealthInformation(healthInfo);
+			await m_client.Services.ReportServiceHealthAsync(
+					serviceId: m_serviceId,
+					healthInformation: sfcHealthInfo
+				).ConfigureAwait(false);
+		}
+
+		/// <summary>
+		/// Just mock
+		/// </summary>
+		/// <param name="report"></param>
+		/// <param name="cancellationToken"></param>
+		/// <returns></returns>
+		public override async Task PublishAsync(HealthReport report, CancellationToken cancellationToken)
+		{
+			try
+			{
+				// We trust the framework to ensure that the report is not null and doesn't contain null entries.
+				foreach (KeyValuePair<string, HealthReportEntry> entryPair in report.Entries)
+				{
+					cancellationToken.ThrowIfCancellationRequested();
+					await PublishHealthInfo(BuildSfHealthInformation(entryPair.Key, entryPair.Value), cancellationToken);
+				}
+
+				cancellationToken.ThrowIfCancellationRequested();
+				await PublishHealthInfo(BuildSfHealthInformation(report), cancellationToken);
+			}
+			catch (Exception)
```

nit: If not using exception can just have empty catch
i.e.
catch
{
    
}